scripts/local_utility.py

```diff
@@ -1,11 +1,37 @@
+import itertools
 from pathlib import Path
 
 from loguru import logger
 
+from app.util import generate_uuid
 
-def gather_valid_files(root: Path, pattern: str = '**/*.*'):
-    for item in root.glob(pattern):
-        if item.suffix in ['.jpg', '.png', '.jpeg', '.jfif', '.webp', '.gif']:
-            yield item
-        else:
-            logger.warning("Unsupported file type: {}. Skip...", item.suffix)
+
+def gather_valid_files(root: Path, pattern: str = '**/*.*', max_files=None):
```

Comment:
Thumbnail generator is using this, we need to either make it return `Path` when `max_files=None` or refactor the thumbnail generator.